src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMetaTileEntity_QuantumForceTransformer.java

```diff
@@ -0,0 +1,1160 @@
+                                    .dot(5)
+                                    .buildAndChain(
+                                            onElementPass(x -> ++x.mCasing, ofBlock(ModBlocks.blockCasings2Misc, 12))))
+                    .build();
+        }
+        return this.STRUCTURE_DEFINITION;
+    }
+
+    public final boolean addCatalystHousing(IGregTechTileEntity aTileEntity, int aBaseCasingIndex) {
+        if (aTileEntity == null) {
+            return false;
+        } else {
+            IMetaTileEntity aMetaTileEntity = aTileEntity.getMetaTileEntity();
+            if (aMetaTileEntity instanceof GT_MetaTileEntity_Hatch_Catalysts) {
+                return addToMachineList(aTileEntity, aBaseCasingIndex);
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public boolean addToMachineList(IGregTechTileEntity aTileEntity, int aBaseCasingIndex) {
+        final IMetaTileEntity aMetaTileEntity = aTileEntity.getMetaTileEntity();
+        if (aMetaTileEntity == null) {
+            return false;
+        }
+        if (aMetaTileEntity instanceof GT_MetaTileEntity_Hatch_Catalysts) {
+            log("Found GT_MetaTileEntity_Hatch_Catalysts");
+            return addToMachineListInternal(mCatalystBuses, aMetaTileEntity, aBaseCasingIndex);
+        }
+        return super.addToMachineList(aTileEntity, aBaseCasingIndex);
+    }
+
+    @Override
+    public boolean checkMachine(IGregTechTileEntity aBaseMetaTileEntity, ItemStack aStack) {
+        this.mCasing = 0;
+        this.mMinimumMufflerTier = 0;
+        mCatalystBuses.clear();
+        if (checkPiece(this.mName, 7, 20, 4)
+                && checkHatch()
+                && (mTecTechEnergyHatches.size() >= 1 || mEnergyHatches.size() >= 1)) {
+            return true;
+        }
+        return false;
+    }
+
+    @Override // The lowest tier muffler of the total 21 will decide the muffler tier in the multi
+    public boolean addMufflerToMachineList(IGregTechTileEntity aTileEntity, int aBaseCasingIndex) {
+        if (aTileEntity == null) {
+            return false;
+        } else {
+            IMetaTileEntity aMetaTileEntity = aTileEntity.getMetaTileEntity();
+            if (aMetaTileEntity instanceof GT_MetaTileEntity_Hatch_Muffler) {
+                if (mMinimumMufflerTier == 0
+                        || ((GT_MetaTileEntity_Hatch_Muffler) aMetaTileEntity).mTier < mMinimumMufflerTier) {
+                    mMinimumMufflerTier = ((GT_MetaTileEntity_Hatch_Muffler) aMetaTileEntity).mTier;
+                }
+                return addToMachineList(aTileEntity, aBaseCasingIndex);
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public void construct(ItemStack stackSize, boolean hintsOnly) {
+        buildPiece(this.mName, stackSize, hintsOnly, 7, 20, 4);
+    }
+
+    @Override
+    public int survivalConstruct(ItemStack stackSize, int elementBudget, IItemSource source, EntityPlayerMP actor) {
```

Comment:
It would be good to migrate to the new survival placement calls, that use the `AutoPlaceEnvironment `.

Reply:
Please go and migrate whole GT++. As I am pretty sure other multis also need to be migrated. Can do it later tho.